### PR TITLE
fixes crash in sign up activity in api 21 and 22

### LINKED ISSUE #5303: [Bug]: App crashes when trying to sign up on lowest supported api 21.
### Summary

Need to update/downgrade google materials library version. I am supposing downgrading won't be an option here, so for upgrading need to update jdk version to at least 17.

### Steps to reproduce

* Try to Sign up on fresh app install.
* The app will crash.

### Expected behaviour

App should not crash

### Actual behaviour

App crashes when trying to open a webview.

### Device name

Pixel 5 emulator

### Android version

api 21

### Commons app version

prodDebug

### Device logs

_No response_

### Screen-shots

_No response_

### Would you like to work on the issue?

None

### PR DESCRIPTION
**Pull request by srishti-R**
**Description (required)**

Fixes #5303 

What changes did you make and why?
Fixes crash fix observed on API 21 when trying to sign up.

**Tests performed (required)**

Tested {prodDebug} on {pixel 5 emulator} with API level {21}.

**Screenshots (for UI changes only)**
signup

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java [hunk 1]
```diff
@@ -61,4 +63,13 @@ public void onBackPressed() {
             super.onBackPressed();
         }
     }
+
+    @Override
+    public void applyOverrideConfiguration(final Configuration overrideConfiguration) {
+        if (Build.VERSION.SDK_INT == 21 || Build.VERSION.SDK_INT == 25 && (
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java:69]: 25 is Nougat right? https://developer.android.com/reference/android/os/Build.VERSION_CODES#N_MR1
Did you mean 22 by any chance?

File: app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java [hunk 2]
```diff
@@ -61,4 +63,13 @@ public void onBackPressed() {
             super.onBackPressed();
         }
     }
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java:66]: Would you mind adding javadoc here explaining why this code is necessary, and linking to https://github.com/commons-app/apps-android-commons/issues/5303? Thanks :-)

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I did not test on lower APIs, but at least I can confirm that on Android 14 the app still works and uploads fine with this pull request. :-)

Would you mind addressing the 2 comments? Thanks a lot!

File: app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java (hunk 1 above)
Comment (srishti-R) [reply app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java]: I am extremely sorry, should have reviewed the code before pushing, this line intended to say for API levels between 21 to 25, have made the correction.

File: app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java (hunk 2 above)
Comment (srishti-R) [reply app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java:66]: added

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/auth/SignupActivity.java:66]: Great doc, very useful to future readers, thanks! :-)

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**